bin/clover/src/cloud-control-funcs/overrides/AWS::EC2::VPNConnection/actions/refresh.ts

```diff
@@ -0,0 +1,91 @@
+async function main(component: Input): Promise<Output> {
+    let name = component.properties?.si?.resourceId;
+    const resource = component.properties.resource?.payload;
+    if (!name) {
+      name = resource.name;
+    }
+    if (!name) {
+      return {
+        status: component.properties.resource?.status ?? "error",
+        message: "Could not refresh, no resourceId present",
+      };
+    }
+  
+    const child = await siExec.waitUntilEnd("aws", [
+      "cloudcontrol",
+      "get-resource",
+      "--region",
+      _.get(component, "properties.domain.extra.Region", ""),
+      "--type-name",
+      _.get(component, "properties.domain.extra.AwsResourceType", ""),
+      "--identifier",
+      name,
+    ]);
+  
+    if (child.exitCode !== 0) {
+      console.log("Failed to refresh cloud control resource");
+      console.log(child.stdout);
+      console.error(child.stderr);
+      return {
+        status: "error",
+        payload: resource,
+        message:
+          `Refresh error; exit code ${child.exitCode}.\n\nSTDOUT:\n\n${child.stdout}\n\nSTDERR:\n\n${child.stderr}`,
+      };
+    }
+  
+    const resourceResponse = JSON.parse(child.stdout);
+    const payload = JSON.parse(
+      resourceResponse["ResourceDescription"]["Properties"],
+    );
+
+    // Check if VPNConnection has a TransitGatewayId
+    const transitGatewayId = _.get(component, "properties.domain.TransitGatewayId");
+
+    // If no TransitGatewayId, return the payload as normal
+    if (!transitGatewayId) {
+        return {
```

Comment:
You can restructure the code to not have so many return paths - just remove this and the return inside the try and it will always fall through to the last return at the end of the file